lib/src/main/scala/spinal/lib/sim/GenerateDiagram.scala

```diff
@@ -0,0 +1,510 @@
+          edges.add(newEdge)
+      }
+    }
+  }
+/** Setting and adding Edge*/
+
+  private def drawNodes(thisNode: ElkNode): Unit = {
+    builder ++= s"""{id:"${thisNode.labelName}",\n"""
+    if (thisNode.typeName != "") builder ++= s"""type:"${thisNode.typeName}",\n"""
+    if (thisNode.highlight != 0) builder ++= s"""highlight:${thisNode.highlight},\n"""
+    if (thisNode.inPorts.nonEmpty) {
+      builder ++= s"""inPorts: ["""
+      for (inPort <- thisNode.inPorts) builder ++= s""""$inPort","""
+      builder ++= s"""],\n"""
+    }
+    if (thisNode.outPorts.nonEmpty) {
+      builder ++= s"""outPorts: ["""
+      for (outPort <- thisNode.outPorts) builder ++= s""""$outPort","""
+      builder ++= s"""],\n"""
+    }
+    if (thisNode.children.nonEmpty) {
+      builder ++= s"""children: [\n"""
+      for (thisChildren <- thisNode.children) drawNodes(thisChildren)
+      builder ++= s"""],\n"""
+    }
+    if (thisNode == topNode) drawEdges()
+    builder ++= s"""},\n"""
+  }
+/** Generating HDElk language for Node*/
+
+
+  private def drawEdges(): Unit = {
+    builder ++= s"""edges:[\n"""
+    for (edge <- edges) {
+      builder ++= s"""{ source:"${edge.source}",target:"${edge.target}",bus:${edge.isBus},"""
+      if (edge.label != "")
+        builder ++= s"""label:"${edge.label}","""
+      if (edge.highlight != 0)
+        builder ++= s"""highlight:${edge.highlight}"""
+      builder ++= s"""},\n"""
+    }
+    builder ++= s"""]\n"""
+  }
+/** Generating HDElk language for Edge*/
+
+  private def drawClockDomains():Unit = {
+    builder ++= s"""{id:"ClockDomains",\nchildren:[\n"""
+    for (element <- clkNamesMap) {
+      builder ++= s"""{id:"${element._1}",highlight:${element._2}},\n"""
+    }
+    builder ++= s"""]\n}\n"""
+  }
+/** Generating HDElk language for Clock legend*/
+  def beginDraw():Unit = {
+    builder ++=
+      s"""
+         |<div id="${topNode.labelName}"></div>
+         |<h3>${topNode.labelName}</h3><br><br><br><br>
+         |<script type="text/javascript">
+         |
+         |var mygraph = {
+         |children:[
+         |""".stripMargin
+    dealAllSignals()
+    drawNodes(topNode)
+    if (clkMap.nonEmpty)
+      drawClockDomains()
+    builder ++= s"""],\n}\nhdelk.layout( mygraph,"${topNode.labelName}");\n</script>\n"""
+    val fileName = topLevelName + ".html"
+    val file = new File(fileName)
+    val pw = new FileWriter(file, true)
+    pw.write(builder.toString())
+    pw.close()
+  }
+}
+/** Integrating all methods to draw an image*/
+
+class GenerateDiagrams(rtl: SpinalReport[Component]) {
+  def beginGenerate(): Unit = {
+    val module = rtl.toplevel
+    val fileName = rtl.toplevelName + ".html"
+    val file = new File(fileName)
+    val pw = new FileWriter(file)
+    val builder = new StringBuilder()
+    builder ++=
+      s"""<!DOCTYPE html>
+         |<html>
+         |<head>
+         |    <meta charset="UTF-8">
+         |    <title>RTL连接图</title>

```

Comment:
Please keep with the remainder of SpinalHDL and use an English title

Reply:
Sorrry,i forgot to change it.I have changed it to "RTL diagrams based on SpinalHDL code"

Reply:
That's better, we could also use the component name of the top component in the title e.g. `f"{rtl.toplevelName} RTL diagrams"`